src/main/java/mate/academy/service/order/impl/OrderServiceImpl.java

```
@@ -0,0 +1,152 @@
+import mate.academy.dto.order.item.OrderItemResponseDto;
+import mate.academy.exception.EntityNotFoundException;
+import mate.academy.exception.OrderProcessingException;
+import mate.academy.mapper.OrderItemMapper;
+import mate.academy.mapper.OrderMapper;
+import mate.academy.model.CartItem;
+import mate.academy.model.Order;
+import mate.academy.model.OrderItem;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.repository.order.OrderRepository;
+import mate.academy.repository.order.item.OrderItemRepository;
+import mate.academy.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.repository.user.UserRepository;
+import mate.academy.service.order.OrderService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public OrderResponseDto createOrder(Long userId, OrderRequestDto request) {
+        User user = findUserById(userId);
+        ShoppingCart cart = getShoppingCartWithValidation(userId);
+
+        Order order = buildOrder(user, request, cart);
+        saveOrderAndClearCart(order, cart);
+
+        return orderMapper.toDto(order);
+    }
+
+    private User findUserById(Long userId) {
+        return userRepository.findById(userId)
+                .orElseThrow(()
+                        -> new OrderProcessingException("User not found with id " + userId));
```

usually it's better to throw conventional EntityNotFoundException in case an entity is not found. Fix in all places